### DISCUSSION TITLE
Deploying on my own server

### DISCUSSION THREAD

**Discussion by ryanhaticus** [vercel/next.js#38713] [Help]
Hi there.

I recently was looking to move away from Vercel, however I still love and want to continue to use the Next.js framework. I'm aware Next ships with a `next start` command to create a production server. My question is, how does Next treat things such as serverless functions in the `/pages/api` directory? Will they work as normal? If so, will I still experience cold starts?

Thanks!

**Comment by icyJoseph** [comment]
> how does Next treat things such as serverless functions in the /pages/api directory? Will they work as normal? If so, will I still experience cold starts?

As far as I know, yes to both, but in the second case, only once, IIRC. Which looks good on surface, but it means there's a handler eating memory that's not needed anymore. 

You can do something like:

```ts
const id = Math.random().toString(36).slice(2) + Date.now().toString(36);
```

And use the `id` in a response payload, then you know which lambda you are talking to, in Vercel for example, you can expect to hit different ones. It is almost like a new thread is spun per request.

However, in self hosting you talk to the same lambda, at least for as long as I've tested, which wasn't very long. In self hosting I guess you'd either try to re-implement Vercel's strategy by hand (might be better to just stick to their service), or use various containers and load balancers that spin up Next.js instances, and shut them down as needed. And then you'll experience cold starts here and there.

It'd be nice to hear from the team or see if you can do some more testing. I am now on vacation but when I come back I'll do some experiments on one app we have, which is hosted in an EC2.
